src/libYARP_serversql/src/NameServiceOnTriples.h

```diff
@@ -57,12 +57,12 @@ class NameServiceOnTriples : public yarp::name::NameService {
     Subscriber *subscriber;
     std::string lastRegister;
     yarp::os::Contact serverContact;
-    yarp::os::Semaphore mutex;
+    yarp::os::Semaphore mutex, access;
     bool gonePublic;
     bool silent;
     yarp::os::NameSpace *delegate;
 public:
-    NameServiceOnTriples() : mutex(1) {
+    NameServiceOnTriples() : mutex(1), access(1) {
```

Comment:
This might be a good chance to also use proper initializers for the other attributes ..